sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -103,13 +106,16 @@ abstract class StringRegexExpression extends BinaryExpression
     Use RLIKE to match with standard regular expressions.
   """,
   since = "1.0.0")
-case class Like(left: Expression, right: Expression) extends StringRegexExpression {
+case class Like(left: Expression, right: Expression, escapeCharOpt: Option[String] = None)
+  extends StringRegexExpression {
 
-  override def escape(v: String): String = StringUtils.escapeLikeRegex(v)
+  private lazy val escapeStr = escapeCharOpt.getOrElse("\\")
```

Comment:
we need this lazy?

Reply:
The `doGenCode` exists two code execution path. when `right.foldable` is false, escapeStr will be not used.

Reply:
duplicated in the line 148?

Reply:
`escapeStr` works fine in line 111, but not normal in line 149. 
If I change to:
`private lazy val escapeStr = escapeCharOpt.getOrElse("""\\""")`
`escapeStr` works fine in line 149, but not normal in line 111.